specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json

```diff
@@ -0,0 +1,1205 @@
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "Specifies the hybrid machine unique ID."
+        },
+        "displayName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the hybrid machine display name."
+        },
+        "machineFqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the hybrid machine FQDN."
+        },
+        "clientPublicKey": {
+          "type": "string",
+          "description": "Public Key that the client provides to be used during initial resource onboarding"
+        },
+        "osName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The Operating System running on the hybrid machine."
+        },
+        "osVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The version of Operating System running on the hybrid machine."
+        },
+        "vmUuid": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the Arc Machine's unique SMBIOS ID"
+        },
+        "extensions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MachineExtensionInstanceView"
+          },
+          "description": "Machine Extensions information"
+        },
+        "osSku": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the Operating System product SKU."
+        },
+        "domainName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the Windows domain name."
+        },
+        "adFqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the AD fully qualified display name."
+        },
+        "dnsFqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the DNS fully qualified display name."
+        },
+        "privateLinkScopeResourceId": {
+          "type": "string",
+          "description": "The resource id of the private link scope this machine is assigned to, if any."
+        }
+      },
+      "description": "Describes the properties of a hybrid machine."
+    },
+    "MachineUpdateProperties": {
+      "properties": {
+        "locationData": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/locationData"
+        },
+        "privateLinkScopeResourceId": {
+          "type": "string",
+          "description": "The resource id of the private link scope this machine is assigned to, if any."
+        }
+      },
+      "description": "Describes the ARM updatable properties of a hybrid machine."
+    },
+    "Machine": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
```

Comment:
these were all in from previous api versions, so I'm unsure where they're actually appropriate. I got the impression from looking into it that we've been using them all wrong, so I removed all of the "allOf" references that only contained a single item. hopefully thats right

Reply:
allOf implies inheritance rather than composition in our generators, so it is appropriate in some cases (models extending Resource), but not in others (places where you want to use a specific type).  In terms of the REST API, these are interchangeable, but they have very different effects for client generation